build/visual-studio/platform/platform.vcxproj

```diff
@@ -27,25 +27,25 @@
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
-    <ConfigurationType>StaticLibrary</ConfigurationType>
+    <ConfigurationType>DynamicLibrary</ConfigurationType>
```

Comment:
I'm a bit nervous about introducing Yet Another DLL, but if this is only impactful to our examples and tests, then this is fine.

In the long term we really need to figure out how we want to factor our code into dynamic libraries for long-term use. My tendency is to assume we want a kind of Slang "core" DLL that includes what amounts to the language "runtime" plus all these platform-abstraction services like window creation (basically the core/stdlib stuff), and then have a separate slang compiler DLL that provides the current public compilation API. It seems like this `platform` code is stuff that basically belongs in the core/stdlib DLL in the long run.

Reply:
The current way means that Slang itself doesn't need to depend on things like XLib, which is nice.